amundsen_application/static/js/components/TableDetail/RequestMetadataForm/index.tsx

```diff
@@ -0,0 +1,129 @@
+import { toggleRequest, submitNotification } from 'ducks/notification/reducer';
+import { bindActionCreators } from 'redux';
+
+interface StateFromProps {
+  userEmail: string;
+  tableName: string;
+  tableOwners: Array<string>;
+  requestIsOpen: boolean;
+}
+
+export interface DispatchFromProps {
+  submitNotification: (
+    recipients: Array<string>,
+    sender: string,
+    notificationType: NotificationType,
+    options?: SendNotificationOptions
+  ) => SubmitNotificationRequest;
+  toggleRequest: () => ToggleRequestAction;
+}
+
+export type RequestMetadataProps = StateFromProps & DispatchFromProps;
+
+interface RequestMetadataState {}
+
+export class RequestMetadataForm extends React.Component<RequestMetadataProps, RequestMetadataState> {
+  public static defaultProps: Partial<RequestMetadataProps> = {};
+
+  constructor(props) {
+    super(props);
+  }
+
+  toggle = () => {
+    this.props.toggleRequest();
+  }
+
+  submitNotification = (event) => {
+    event.preventDefault();
+    const form = document.getElementById("RequestForm") as HTMLFormElement;
+    const formData = new FormData(form);
+    const recipientString = formData.get('recipients') as string
+    const recipients = recipientString.split(",")
+    const sender = formData.get('sender') as string;
+    const descriptionRequested = formData.get('table-description') === "on" ? true : false;
+    const fieldsRequested = formData.get('column-description') === "on" ? true : false;
+    const comment = formData.get('comment') as string;
+    this.props.submitNotification(
+      recipients,
+      sender,
+      NotificationType.REQUESTED,
+      {
+        comment,
+        resource_name: this.props.tableName,
+        resource_url: window.location.href,
+        description_requested: descriptionRequested,
+        fields_requested: fieldsRequested,
+      }
+    )
+  };
+
+  render() {
+    const expandedClass = this.props.requestIsOpen ? 'expanded' : 'collapsed';
+    return (
+      <div className={`request-component ${expandedClass}`}>
+        <div className="form-section request-header">
+          <h3 className="title">{TITLE_TEXT}</h3>
+          <button type="button" className="btn btn-close" aria-label={"Close"} onClick={this.toggle}/>
+        </div>
+        <form onSubmit={ this.submitNotification } id="RequestForm">
+          <div className="form-section">
+            <label>{FROM}</label>
+            <input type="email" name="sender" className="form-control" required={true} value={this.props.userEmail}/>
+          </div>
+          <div className="form-section">
+            <label>{TO}</label>
+            <input type="email" name="recipients" className="form-control" required={true} multiple={true} defaultValue={this.props.tableOwners.join(",")}/>
+          </div>
+          <div className="form-section">
+            <label>{REQUEST_TYPE}</label>
+            <p><input type="checkbox" name="table-description"/> {TABLE_DESCRIPTION}</p>
```

Comment:
Generally we want the checkbox description to be clickable. There are two main ways of doing this. Either way we want to use the `<label>` element and not the `<p>`

1) The `for` attribute should match the input box's `id` attribute
2) The `label` element wraps the `input` element and is automatically connected (recommended).